### PR TITLE
Switch to github actions

### PR DESCRIPTION
**Pull request by milessabin**
Also update Scala Native to 0.4.0.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - uses: olafurpg/setup-gpg@v3
+
+      - env:
+          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+        run: sbt ++${{ matrix.scala }} ci-release
```
Comment (joroKr21) [inline .github/workflows/ci.yml:111]: Will this publish all Scala versions? It looks to me like it won't.

**Review by joroKr21 — APPROVED** [review]

**Comment by joroKr21** [comment]
> /cc @xuwei-k @lolgab this will supercede #1073 and #1077.

not quite, we still need to dig out a 2.3.3 binary compatible version 😄

File: .github/workflows/ci.yml (hunk 1 above)
Comment (joroKr21) [reply .github/workflows/ci.yml:111]: Ok I see we did `sbt ci-release` before as well.
`++${{ matrix.scala }}` confused me 🙈

**Comment by milessabin** [comment]
> not quite, we still need to dig out a 2.3.3 binary compatible version :smile: 

Agreed, but I think this is a good place to start from :smiley_cat:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (milessabin) [reply .github/workflows/ci.yml:111]: To be completely honest, I'm not sure. I want to merge this and see how it behaves :shrug:

I thought https://github.com/djspiewak/sbt-github-actions/pull/56 looked like a reasonable complaint too ... let's see how it goes :smile:

**Pull request merged**